test/rules/always_put_control_body_on_new_line.dart

```diff
@@ -19,6 +19,13 @@ testIfElse() {
     return; // OK
   else return; // LINT
 
+  if (false) {
+  }
+  else if (false) { // OK
```

Comment:
That's why I qualified it with "as much as possible". I still think the example I gave should not produce any lints and that it would be good to have a test case to verify that, but it obviously isn't critical (and your changes were already merged, so it's a rather moot point).